app/modules/mdns.c

```diff
@@ -65,28 +67,39 @@ static wrapper_t *process_args(lua_State *L, phase_t phase, size_t *sizep)
   }
 
   if (phase == PHASE_CALCULATE_LENGTH) {
-    luaL_checktype(L, 1, LUA_TSTRING);
-    luaL_checktype(L, 2, LUA_TSTRING);
-    (void) luaL_checkinteger(L, 3);
-    *sizep += c_strlen(luaL_checkstring(L, 1)) + 1;
-    *sizep += c_strlen(luaL_checkstring(L, 2)) + 1;
+    luaL_checktype(L, arg_no, LUA_TSTRING);
```

Comment:
This isn't the standard Lua way of doing this, so I don't understand the rationale here.  For example, I would expect these lines to be something like:
```C
    host = luaL_checklstring (L, arg_no++, &hostlen);
    service = luaL_checklstring (L, arg_no++, &servicelen);
    if (lua_type(L, 4) == LUA_NUMBER) {
      service_desc = luaL_checklstring (L, arg_no++, &service_desclen);
    } else {
      service_desc = NULL;
      service_desclen = 0;
    }
    port_no = luaL_checkinteger(L, arg_no++);
```
etc.